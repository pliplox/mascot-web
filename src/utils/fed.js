export const getUserName = () => "juan perez";

export const getHour = () => {
  const date = new Date();
  let hourAndMinutes = `${date.getHours()}:${date.getMinutes()}`;
  return hourAndMinutes;
};

export const getToday = () => {
  let day = new Date().getDay();
  if (day === 0) {
    day = 6;
  }
  return day;
};
