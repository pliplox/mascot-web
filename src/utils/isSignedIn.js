/**
 * Verifies wether or not the user is signed in the system.
 * @return {Boolean} True if user is signed in, false otherwise.
 */
const isSignedIn = () => {
  const tokenId = localStorage.getItem('tokenId');
  return tokenId !== undefined && tokenId !== null;
};

export default isSignedIn;
