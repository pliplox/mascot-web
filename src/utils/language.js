export default [
  { name: 'Espa√±ol', code: 'es' },
  { name: 'English', code: 'en' }
];
