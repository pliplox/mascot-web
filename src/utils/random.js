// eslint-disable-next-line import/prefer-default-export
export const randomAnimalImage = (breakpoint = 'lg') => {
  const breakPoints = {
    lg: '1000/720',
    md: '960/691',
    sm: '600/432',
    xs: '320/570'
  };

  const pickedSize = breakPoints[breakpoint];

  const sources = [
    `https://placebear.com/${pickedSize}`,
    `http://placeimg.com/${pickedSize}/animals`,
    `https://placedog.net/${pickedSize}?random`,
    `https://placekitten.com/${pickedSize}`
  ];

  return `url(${sources[Math.floor(Math.random() * sources.length)]})`;
};
