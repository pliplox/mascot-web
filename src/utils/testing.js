import React from "react";
import { render } from "@testing-library/react";
import { SnackbarProvider } from "notistack";
import { MuiThemeProvider } from "@material-ui/core";
import theme from "./themeMui";

export const renderWithProvider = ({ ...children }) => {  
  return render(
    <MuiThemeProvider theme={theme}>
      <SnackbarProvider>{children}</SnackbarProvider>
    </MuiThemeProvider>
  );
};
