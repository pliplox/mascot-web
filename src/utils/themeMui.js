import { createMuiTheme } from "@material-ui/core";
import { indigo } from "@material-ui/core/colors";

const theme = createMuiTheme({
  palette: {
    secondary: {
      main: indigo.A100,
      light: indigo[50],
      dark: indigo[500],
    },
    primary: {
      main: "#460151",
      light: "#42e8dc",
      dark: "#380250",
      white: '#D4D4D4'
    },
  },
  typography: {
    body1: {
      fontWeight: 500,
    },
  },
  overrides: {
    MuiContainer: {
      maxWidthXl: {
        paddingTop: "1vh",
        paddingLeft: "5vw",
        paddingRight: "5vW",
        paddingBottom: "5vW",
      },
    },
    MuiTypography: {
      colorTextSecondary: "#2196f3",
    },
  },
  spacing: (value) => value ** 2,
});

export default theme;
