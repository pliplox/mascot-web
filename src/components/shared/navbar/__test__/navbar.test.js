import React from "react";
import { fireEvent } from "@testing-library/react";
import { Navbar } from "./../";

import { renderWithProvider, tokenMock } from "../../../../utils/testing";

let wrapper;

describe("Navbar", () => {
  it("show render navbar", () => {
    expect(renderWithProvider(<Navbar />));
  });

  beforeEach(() => {
    localStorage.setItem("tokenId", tokenMock());
    wrapper = renderWithProvider(<Navbar />);
  });

  it("show the app title", () => {
    const { getByText } = wrapper;
    expect(getByText("MascotApp")).toBeInTheDocument();
  });

  it("show the icon profile", () => {
    const { getByTestId } = wrapper;
    expect(getByTestId("icon-profile")).toBeInTheDocument();
  });

  it("show open profile menu", () => {
    const { getByTestId, getByText } = wrapper;
    fireEvent.click(getByTestId("icon-profile"));
    expect(getByText("Profile")).toBeInTheDocument();
    expect(getByText("My account")).toBeInTheDocument();
    expect(getByText("Logout")).toBeInTheDocument();
  });

  it("show the icon drawer", () => {
    const { getByTestId } = wrapper;

    expect(getByTestId("drawer")).toBeInTheDocument();
  });

  describe("When drawer is clicked", () => {
    let drawerSettings;
    beforeEach(() => {
      fireEvent.click(wrapper.getByTestId("drawer"));
      drawerSettings = wrapper.getByText("Configuracion");
    });

    it("shows sideBar elements", () => {
      const { getByText } = wrapper;

      expect(getByText("Alimentar")).toBeInTheDocument();
      expect(getByText("Grupo")).toBeInTheDocument();
      expect(drawerSettings).toBeInTheDocument();
    });

    describe("When drawer settings is clicked", () => {
      beforeEach(() => {
        fireEvent.click(drawerSettings);
      });
      it("show drawer settings", () => {
        const { getByText } = wrapper;
        expect(getByText("TimeZone")).toBeInTheDocument();
      });
    });
  });
});
