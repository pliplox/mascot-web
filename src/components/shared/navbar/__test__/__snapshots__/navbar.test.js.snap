// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render navbar 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-grow-1"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <button
          aria-label="open drawer"
          class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-2 MuiIconButton-colorInherit MuiIconButton-edgeStart"
          data-testid="drawer"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <h6
          class="MuiTypography-root makeStyles-title-4 MuiTypography-h6 MuiTypography-noWrap"
        >
          Skoll Web
        </h6>
        <div
          class="makeStyles-grow-1"
        />
        <div>
          <button
            aria-controls="primary-search-account-menu"
            aria-haspopup="true"
            aria-label="account of current user"
            class="MuiButtonBase-root MuiIconButton-root makeStyles-accountMenu-3 MuiIconButton-colorInherit MuiIconButton-edgeEnd"
            data-testid="icon-profile"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </header>
  </div>
</DocumentFragment>
`;
